src/eureka/S5_lightcurve_fitting/differentiable_models/StarryModel.py

```diff
@@ -114,8 +114,80 @@ def setup(self, newparams):
             pl_params = PlanetParams(self, 0, chan, eval=False)
 
             # Initialize star object
-            star = starry.Primary(starry.Map(udeg=self.udeg),
-                                  m=0, r=pl_params.Rs)
+            # check for spots and set parameters
+            if hasattr(self.parameters, 'spotrad0'):
+                # find number of spots
+                spotinds = [s for s in self.parameters.dict.keys() 
+                            if 'spotrad' in s]   
+                counter = [s for s in spotinds if '_' in s]
+                nspots = len(spotinds) - len(counter)
+
+                # create arrays to hold values
+                spotrad = np.zeros((self.nchannel_fitted, nspots))
+                spotlat = np.zeros((self.nchannel_fitted, nspots))
+                spotlon = np.zeros((self.nchannel_fitted, nspots))
+                spotcon = np.zeros((self.nchannel_fitted, nspots))
+                starrot = np.zeros((self.nchannel_fitted))
+                starinc = np.zeros((self.nchannel_fitted))
```

Comment:
updated these arrays to be the defaults instead of just blank arrays of zeros in all locations (I assumed they'd always be overwritten, but if someone forgets to set one it makes sense to have it be the default)